### PR TITLE
Run tests for code written in TypeScript

### PR DESCRIPTION
**Pull request by arkflpc**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Feature (tests): Added support for running automated tests for the TypeScript code. Closes ckeditor/ckeditor5#11716.

---

### Additional information

*For example – encountered issues, assumptions you had to make, other affected tickets, etc.*

### CODE REVIEW

**Review by psmyrek — APPROVED** [review]
LGTM 🎉 

✅ code coverage
✅ file watcher
✅ source maps
✅ debugging with breakpoints and conditional breakpoints in Chrome DevTools
✅ `CK_DEBUG` and `CK_DEBUG_*` flags support

**Comment by psmyrek** [comment]
BTW, this PR targets only automated tests. For manual test, there is a dedicated issue: https://github.com/ckeditor/ckeditor5/issues/11717.

I've also checked the following cases:
✅ source file is TypeScript
✅ source file is TypeScript, but it imports JavaScript
✅ source file is JavaScript, but it imports TypeScript
✅ source file is JavaScript, but it imports JavaScript

File: packages/ckeditor5-dev-tests/lib/utils/automated-tests/getwebpackconfig.js [hunk 1]
```diff
@@ -76,6 +80,19 @@ module.exports = function getWebpackConfigForAutomatedTests( options ) {
...
+					use: [
+						{
+							loader: require.resolve( '../ck-debug-loader' ),
+							options: {
+								debugFlags: options.debug
+							}
+						},
+						'ts-loader'
+					],
+					exclude: /node_modules/
```
Comment (pomek) [inline packages/ckeditor5-dev-tests/lib/utils/automated-tests/getwebpackconfig.js:95]:
```suggestion
					]
```

As pointed out ☝️

**Review by pomek — APPROVED** [review]
LGTM.

**Pull request merged**